models/organization/org.go

```diff
@@ -650,12 +658,19 @@ func (org *Organization) getUserTeamIDs(ctx context.Context, userID int64) ([]in
 	return teamIDs, db.GetEngine(ctx).
 		Table("team").
 		Cols("team.id").
-		Where("`team_user`.org_id = ?", org.ID).
-		Join("INNER", "team_user", "`team_user`.team_id = team.id").
-		And("`team_user`.uid = ?", userID).
+		Where(builder.In("team.id", userTeamIDbuilder(org.ID, userID))).
```

Comment:
Looks like it's unnecessary `select id from team where id in (select id where ...`.

Reply:
yes, but DBMS should optimize it out ... @lunny if you know how to have the function creating the builder query, to be used, while remove one select ... just tell me :)

Reply:
Why not revert to original version? Looks like `getUserTeamIDsQueryBuilder` is used in only 1 place. I don't think it's necessary to extract it as a new function.

Reply:
it will be used in https://github.com/go-gitea/gitea/pull/32211 but I'm fine with having two implementations (one for the direct query and one for generate the builder) ...